OpenSim/Analyses/OutputReporter.cpp

```diff
@@ -0,0 +1,129 @@
+/* -------------------------------------------------------------------------- *
+ *                        OpenSim: OutputReporter.cpp                         *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2017 Stanford University and the Authors                *
+ * Author(s):                                                                 *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+
+//=============================================================================
+// INCLUDES
+//=============================================================================
+#include "OutputReporter.h"
+#include <OpenSim/Simulation/Model/Model.h>
+
+using namespace OpenSim;
+using namespace std;
+
+//=============================================================================
+// ANALYSIS
+//=============================================================================
+int OutputReporter::begin(SimTK::State& s)
+{
+    if (!proceed()) return(0);
+
+    _model = _model->clone();
```

Comment:
Assigning back to the same variable is confusing to me. It seems that you're leaking this model. I think it would be better to use a `SimTK::ResetOnCopy<std::unique_ptr<Model>>` or a `SimTK::ClonePtr` member variable.

Reply:
👍